### ISSUE TITLE
1.8.0 release

### ISSUE THREAD

**Issue by jorisvandenbossche** [Toblerity/Shapely#1066]
I think it would be good to start planning a 1.8.0 release, so opening an issue to track this (@sgillies).

Remaining issues tagged with the milestone: https://github.com/Toblerity/Shapely/milestone/37 

(there are a few older feature requests that haven't seen any movement on the milestone; those can probably be moved to a next milestone / removed from the milestone?)

Regarding the deprecations and changes in preparation of the Shapely 2.0 RFC:

- Deprecations are mostly done (tracked in https://github.com/Toblerity/Shapely/issues/932), we still need to decide on the STRtree (https://github.com/Toblerity/Shapely/pull/1064)
- I still need to add a good overview of those deprecations to the changelog

**Comment by sgillies** [comment]
@jorisvandenbossche thank you for opening this ticket! I have time to update the change log for 1.8.0 and be the release manager. It's on my Mapbox team's roadmap and everything :smile:

**Comment by sgillies** [comment]
Today I made a pass through the issues to close ones that will obviously be made obsolete by 2.0 (involving ctypes and dynamic GEOS loading) or involve old versions of GEOS. Shapely 1.8.0 will not address any of those issues, it's only about deprecation warnings and critical bug fixes. Whew!

**Comment by jorisvandenbossche** [comment]
That was a nice flood of notifications! (in a positive sense, to be clear ;-), it's good to clean those up)

I will try to write some changelog content about the deprecations in the coming days.

For me, one main item that I would still like to see resolved before doing 1.8 is the discussion about the STRtree design in https://github.com/Toblerity/Shapely/pull/1064

**Comment by sgillies** [comment]
I've added deprecation notices to the change log in 3c70019.

I'm inclined to make a 1.8a1 without resolving the STRtree discussion, but will add some vague deprecation warning about upcoming changes. Doing this will allow us to get feedback on other changes while we resolve the STRtree issue.

**Comment by jorisvandenbossche** [comment]
That sounds good to me

**Comment by sgillies** [comment]
1.8a1 wheels with GEOS 3.9.1 are building successfully. Will upload later today using my phone as a mobile hotspot because of a cable outage at my home office.

**Comment by sgillies** [comment]
:speaking_head: https://pypi.org/project/Shapely/1.8a1/#files

**Comment by RouquinBlanc** [comment]
Are you planning to drop py27 support, or will the corresponding wheel follow?

**Comment by mwtoews** [comment]
Python 2.7 is no longer maintained, so it was dropped for shapely 1.8 (see #743 / #884)

But if you still need to use Python 2.7, use the shapely 1.7 series.

**Comment by jorisvandenbossche** [comment]
I made the alpha release also available on conda-forge, can be installed with:

```
conda install -c conda-forge/label/shapely_dev shapely=1.8a1
```

**Comment by sgillies** [comment]
Let's get ready for 1.8a2. I think consensus on the STRtree is the major thing to iron out before we have another alpha release. We have #1064 and #1112 to compare. Maybe we could get 1.8a2 out the week of April 5?

**Comment by jorisvandenbossche** [comment]
Sounds good!

**Comment by sgillies** [comment]
Finally I think we're ready for a 1.8a2. Any objections to my uploading 1.8a2 tomorrow @jorisvandenbossche @mwtoews ?

**Comment by sgillies** [comment]
It's going to slip to Thursday, 2021-07-15.

**Comment by sgillies** [comment]
https://pypi.org/project/Shapely/1.8a2/#files

**Comment by sgillies** [comment]
Let's have a 1.8a3 on 2021-08-23 to get the changes in #1177 out there for some early testing.

**Comment by jorisvandenbossche** [comment]
Sounds good! 
In preparation of a final 1.8 release, I also (finally) started to write a bit more detailed guide on the changes/warnings -> https://github.com/Toblerity/Shapely/pull/1180

**Comment by sgillies** [comment]
https://pypi.org/project/Shapely/1.8a3/#files

New: manylinux2014 wheels for linux arm64 and pythons 3.8 and 3.9.

**Comment by jorisvandenbossche** [comment]
Thoughts about getting towards a final Shapely 1.8.0 release? 

Feature / deprecation wise, I think we are more or less ready for a final release (and I want to do a push on shapely 2.0, for which it would be good to have 1.8 out first).  
I don't know if we have been getting much testing with the alpha releases (although in geopandas we are constantly testing against Shapely master, so that's at least some testing). In practice (unfortunately) we will probably only see more feedback when the release gets out. So maybe we can do one more release candidate? (I don't know if a beta needed before doing an rc?) And of course be ready to do 1.8.x releases to follow-up on potential issues.

---

Taking a look at the issues currently still listed in the 1.8 milestone (https://github.com/Toblerity/Shapely/milestone/37):

There is one remaining PR with an additional deprecation (https://github.com/Toblerity/Shapely/issues/932):

- https://github.com/Toblerity/Shapely/pull/1183


Upstream GEOS issues that will be fixed in Shapely 1.8 wheels by using the latest GEOS version (-> no direct action required here):

- https://github.com/Toblerity/Shapely/issues/767
- https://github.com/Toblerity/Shapely/issues/1098

Bugs in deprecated proxy geometries (-> IMO those are not important to have fixed before doing a final 1.8 release since it's for deprecated features, and are not regressions AFAIK):

- https://github.com/Toblerity/Shapely/issues/1100
- https://github.com/Toblerity/Shapely/issues/1149

The remaining listed issues that might actually require some action:

- https://github.com/Toblerity/Shapely/issues/1179 -> actual bug that could still be fixed
- https://github.com/Toblerity/Shapely/issues/889 -> deprecate `almost_equals`?
- https://github.com/Toblerity/Shapely/issues/860 -> open question about how to handle GeometryCollections in overlay operations
- https://github.com/Toblerity/Shapely/issues/1184 -> the fix is to reorder some import?

None of those sound as blockers to me, but I can also try to help with some of them if we would prefer to include them.

**Comment by jorisvandenbossche** [comment]
(gentle ping for this)

**Comment by sgillies** [comment]
@jorisvandenbossche I agree that none of those are blockers. And since we don't consider any of them to be, we should go directly to a release candidate. How about Monday, 2021-10-04?

**Comment by jorisvandenbossche** [comment]
Sounds good!

**Comment by sgillies** [comment]
Alright, having fixed #1184 I will tag 1.8rc1, and after upgrading GEOS to 3.9.1 for Windows in shapely-wheels, will build and upload.

**Comment by sgillies** [comment]
https://pypi.org/project/Shapely/1.8rc1/#files

**Comment by sgillies** [comment]
No news is good news! I'm going to set the date for 1.8.0 as 2021-10-25.
